webrtc.html

```diff
@@ -11591,7 +11591,76 @@ <h3>DTMF Example</h3>
   console.log('DTMF function not available');
 }
       </pre>
-    </section><!--
+    </section>
+    <section>
+      <h3>Perfect Negotiation Example</h3>
+      <p>Perfect negotiation is a recommended pattern to abstract away signaling
+      from the rest of an application. It lets applications operate on both
+      peer connection objects at once, on either end of the call, without having
+      to worry about signaling state, racing, or glare.</p>
+      <p>It designates different roles to the two peers, with behavior to
+      resolve signaling collisions between them:</p>
+      <ol>
+        <li>
+          <p>The <dfn>polite peer</dfn> uses rollback to avoid collision with an
+          incoming offer</p>
+        </li>
+        <li>
+          <p>The <dfn>impolite peer</dfn> ignores an incoming offer when this
+          would collide with its own</p>
+        </li>
+      </ol>
+      <p>Together, they manage signaling for the rest of the application in a
+      manner that doesn't deadlock.</p>
```

Comment:
Suggest adding: "Which peer has which role needs to be determined before signaling starts, and is not covered in this example."